htlcswitch/switch.go

```diff
@@ -245,8 +246,14 @@ type Switch struct {
 	// This will be retrieved by the registered links atomically.
 	bestHeight uint32
 
-	wg   sync.WaitGroup
-	quit chan struct{}
+	// TODO(yy): remove handleLocalResponseWG, once handleLocalResponse runs
+	// without a goroutine. Currently we can't run handleLocalResponse in
+	// gm, since if gm is stopping, the goroutine won't start and it is
+	// unclear if it safe to skip handleLocalResponse.
+	handleLocalResponseWG sync.WaitGroup
```

Comment:
I don't believe this is necessary as I believe that the composition of waitgroups is equivalent to the waitgroup of the composition of threads when the wait conditions are always called in conjunction with one another.

Reply:
We discussed this offline.

The main reason for the special handling of `handleLocalResponse` was that, in the old version, it was launched unconditionally (even when the switch was stopping), and switching to `GoroutineManager` introduces a change in behavior. However, we need to ensure that the effects are idempotent to prevent inconsistent states in the event of power failures. If this requirement is met, the behavior change should not pose an issue.

Therefore, I removed the special treatment of `handleLocalResponse`, along with the associated `WaitGroup` and TODOs. It is now managed entirely by `GoroutineManager`.